deploy/setup/scripts/install-docker.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# shellcheck disable=SC1091
+source "$SCRIPTS_DIR/_logging.sh"
+
+debug "Checking for Docker..."
+
+if command -v docker >/dev/null 2>&1; then
+  exit 0
+fi
+
+debug "Installing Docker (using distro packages)..."
+
+if command -v apt-get >/dev/null 2>&1; then
+  # Debian/Ubuntu: install Docker from official Docker APT repo for consistent versioning
+
+  # Detect OS and codename
+  OS_ID="$(. /etc/os-release 2>/dev/null; echo "${ID:-debian}")"
+  CODENAME="$(. /etc/os-release 2>/dev/null; echo "${VERSION_CODENAME:-}")"
+  if [ -z "$CODENAME" ] && command -v lsb_release >/dev/null 2>&1; then
+    CODENAME="$(lsb_release -cs 2>/dev/null || true)"
+  fi
+  if [ -z "$CODENAME" ]; then
+    echo "❌ Could not determine distro codename" >&2
+    exit 1
+  fi
+
+  run "echo 'Using Docker APT repo for $OS_ID ($CODENAME)'"
+  run "sudo apt-get update -y"
+  run "sudo apt-get install -y --no-install-recommends ca-certificates curl gnupg"
+  run "sudo install -m 0755 -d /etc/apt/keyrings"
+  run "curl -fsSL https://download.docker.com/linux/$OS_ID/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg"
```

Comment:
I'd prefer to use `docker.io` from the ubuntu / debian repos. Is there a reason to not?

Reply:
I preferred this as I was sure to get the more recent versions and the same code runs regardless of dist/version.

Reply:
I have the strong opinion that it is not good practice to get packages from external sources if distro repos provide what is needed. Is `docker.io` too old in any of the distro repos?

Reply:
Yes, my understanding from you and from testing was that at least one of the Debian repos for one of the supported Debian versions was out of date (only providing docker 1.x)

Reply:
Yes, docker-compose was the problem on debian. Now with trixie being stable, we also have compose v2 natively on debian https://packages.debian.org/search?suite=trixie&arch=any&searchon=names&keywords=docker-compose